mirrord-config/src/util.rs

```diff
@@ -0,0 +1,99 @@
+use std::{slice::Join, str::FromStr};
+
+use serde::Deserialize;
+use thiserror::Error;
+
+pub trait MirrordConfig {
+    type Generated;
+
+    fn generate_config(self) -> Result<Self::Generated, ConfigError>;
+}
+
+impl<T> MirrordConfig for Option<T>
+where
+    T: MirrordConfig + Default,
+{
+    type Generated = T::Generated;
+
+    fn generate_config(self) -> Result<Self::Generated, ConfigError> {
+        self.unwrap_or_default().generate_config()
+    }
+}
+
+pub trait MirrordFlaggedConfig: MirrordConfig + Default {
```

Comment:
Does "Flagged" in this context mean "toggleable" or "optional"?

Reply:
The togglable one, referring to a field wrapped in `FlagField` meaning allow to set as false and not specifying each field as disabled in config file